computestorage/helpers.go

```diff
@@ -0,0 +1,185 @@
+package computestorage
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/Microsoft/go-winio/pkg/security"
+	"github.com/Microsoft/go-winio/vhd"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+const defaultVHDXBlockSizeInMB = 1
+
+// SetupContainerBaseLayer is a helper to setup a containers scratch. It
+// will create and format the vhdx's inside and the size is configurable with the sizeInGB
+// parameter.
+//
+// `layerPath` is the path to the base container layer on disk.
+//
+// `baseVhdPath` is the path to where the base vhdx for the base layer should be created.
+//
+// `diffVhdPath` is the path where the differencing disk for the base layer should be created.
+//
+// `sizeInGB` is the size in gigabytes to make the base vhdx.
+func SetupContainerBaseLayer(ctx context.Context, layerPath, baseVhdPath, diffVhdPath string, sizeInGB uint64) (err error) {
+	var (
+		hivesPath  = filepath.Join(layerPath, "Hives")
+		layoutPath = filepath.Join(layerPath, "Layout")
+	)
+
+	// We need to remove the hives directory and layout file as `SetupBaseOSLayer` fails if these files
+	// already exist. `SetupBaseOSLayer` will create these files internally. We also remove the base and
+	// differencing disks if they exist in case we're asking for a different size.
+	if _, err := os.Stat(hivesPath); err == nil {
+		os.RemoveAll(hivesPath)
```

Comment:
should we worry about any potential issues here? or paths validity/permissions will happen before calling this? I might be off here and those concerns are not relevant at all.

Reply:
Could you elaborate? I'm not fully following

Reply:
we're ignoring the error from `os.RemoveAll`. I took a quick look and some of them are (e.g.) handle wrong paths (with dots in them as one of the cases) and potentially wrong permissions (I think). Is this something we should worry about? for example we're unable to remove something due to wrong path/permissions and we proceed and potentially run into something unexpected?

Reply:
Ahh gotcha gotcha. We can check the error for RemoveAll as we're going to error out in HCS if they couldn't remove anyways, better to just fail earlier. Thanks for the clarification!